juriscraper/state/texas/common.py

```diff
@@ -0,0 +1,786 @@
+        if len(parties) == 0:
+            return name_part
+        if len(parties) == 1:
+            if len(parties[0]["name"]) < len(name_part):
+                return parties[0]["name"]
+            return name_part
+        semi = name_part.find(";")
+        if semi > 0:
+            return name_part[:semi]
+
+        party_indices = [
+            self._find_party_in_case_name(party["name"]) for party in parties
+        ]
+        party_indices.sort(key=lambda x: x[0])
+        start, end = next(
+            indices for indices in party_indices if indices[0] >= 0
+        )
+        return self.case_name_full[start:end]
+
+    @cached_property
+    def case_name_full(self) -> str:
+        """
+        The unshortened name of the case as it appears on TAMES.
+        """
+        name_part_1 = self.case_data["style"]
+        name_part_2 = self.case_data["v"]
+        if len(name_part_2) == 0:
+            return harmonize(name_part_1)
+        return harmonize(f"{name_part_1} v. {name_part_2}")
+
+    @cached_property
+    def case_name(self) -> str:
+        """
+        A (possibly) shortened version of the case name created using some
+        heuristics. Useful to make cases easier to search and match.
+        """
+        name_part_1 = self.case_data["style"]
+        name_part_2 = self.case_data["v"]
+        if len(name_part_2) == 0:
+            return harmonize(name_part_1)
+
+        grouped_parties = {}
+        for k, g in groupby(self.parties, lambda party: party["type"]):
+            if k in grouped_parties:
+                grouped_parties[k].extend(list(g))
+            else:
+                grouped_parties[k] = list(g)
+
+        first_parties = list(
+            chain(
+                grouped_parties.get("Petitioner", []),
+                grouped_parties.get("Appellant", []),
+            )
+        )
+        second_parties = list(
+            chain(
+                grouped_parties.get("Respondent", []),
+                grouped_parties.get("Appellee", []),
+            )
+        )
+
+        name_short_part_1 = self._make_short_party_name(
+            name_part_1, first_parties
+        )
+        name_short_part_2 = self._make_short_party_name(
+            name_part_2, second_parties
+        )
+        return harmonize(f"{name_short_part_1} v. {name_short_part_2}")
+
+    def _parse_docket_number(self) -> str:
+        """
+        Extracts the docket number from the HTML tree. Will fail if
+        `_parse_text` has not yet been called.
+
+        :raises ValueError: If the docket number format is not recognized.
+
+        :return: Docket number.
+        """
+        docket_number = self.case_data["case"]
+        for docket_number_regex in DOCKET_NUMBER_REGEXES:
+            if docket_number_regex.fullmatch(docket_number):
+                return docket_number
+        raise ValueError(f"Unrecognized docket number format: {docket_number}")
+
+    def _parse_date_filed(self) -> datetime:
```

Comment:
Can we return a `date` instead of a `datetime`, since we don’t have the time info?